### DISCUSSION TITLE
online and offlince sync with moor and flutter?

### DISCUSSION THREAD

**Discussion by CodingArcher** [simolus3/drift#1377] [Q&A]
Hey guys,
is it possible to use moor to have a local DB and then sync it with an online SQL DB when internet connection is aviable?

I realy would like to use moor due to it being aviable on all platforms and also being SQl format

**Comment by simolus3** [comment]
Hi,

there are issues related to this topic, e.g. https://github.com/simolus3/moor/discussions/1362, https://github.com/simolus3/moor/issues/1285 or #136. The short answer is that moor does not provide builtin support for this. You can try to implement it yourself by keeping track of the synchronization state (for instance with triggers writing each change into a special log table that you push to the server when an internet connection is available).

**Comment by aldycool** [comment]
Hi,

If you're still interested about synchronization, I've built a synchronization framework that runs on top of Moor library called `netcoresync_moor`. You can find the [repository here](https://github.com/aldycool/NETCoreSync/tree/master/netcoresync_moor) and the [pub.dev package here](https://pub.dev/packages/netcoresync_moor). This framework allows you to have local Moor databases on each user's device and synchronize them to a centralized database hosted on the server.

This framework actually was built for Xamarin technology a couple years ago, and recently ported to Flutter because our team has started moving to Flutter for building our products. And we've chosen to built it on top of Moor (after having insightful discussions with @simolus3, thanks!) because I think Moor is the perfect choice for proven database framework in Flutter right now. Give it a try and feel free to ask questions, thanks.
